### ISSUE TITLE
Fix FP S1764 (`no-identical-expressions`): using BigInts in bit shifting

### ISSUE THREAD

**Issue by Darker-Ink** [SonarSource/eslint-plugin-sonarjs#399]
**I want to report a bug.**

When you are using the `no-identical-expressions` rule and using bigints it seems to cause it to error out when it probably shouldn't

when you do for example

```js
1n << 1n
```

It errors out though when you do

```js
1 << 1
```

it works fine

**Reproducer**

```javascript
const Flags = {
  StaffBadge: 1n << 0n,
  GhostBadge: 1n << 1n, // False-positive here
  SponsorBadge: 1n << 2n
};
```

**Expected behavior:**

It does not error out

**eslint-plugin-sonarjs version:** 0.19.0
**eslint version:** 8.39.0
**Node.js version:** v19.3.0

**Rule key:** no-identical-expressions

**Comment by ilia-kebets-sonarsource** [comment]
Thanks for notifying us, we'll fix this in the next release!
